Minor Windows CMake QoL improvements

shermp opened the pull request:
# Description

Just a couple of minor improvements to make using cmake with the Visual Studio generator nicer on Windows.

1. Add `_CRT_SECURE_NO_WARNINGS` to silence all the warnings about using "insecure" C functions.
2. Enable `UseMultiToolTask` in the Windows cmake presets to use MSBuild Multi Tool Task for much faster build times.

# Manual testing

The change has been manually tested on:

- [x] Windows
- [ ] macOS
- [ ] Linux

1. Configure cmake on the command line (in one of the Visual Studio developer command prompts) or in VS with one of the windows cmake presets
2. Build. Notice the much reduced warnings, and much faster build times

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [x] my change has been manually tested on Windows, macOS, and Linux.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] provided the release notes draft (for significant user-facing changes).
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

johnnovak commented on CMakeLists.txt:
```diff
@@ -202,6 +202,11 @@ target_include_directories(
+# Disable cl/clangcl warnings about insecure C functions
+if(MSVC)
+  add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+endif()
```
Can we move this block to line 28, please?

shermp replied:
Sure, I think so.

It was originally there, because I first tried `target_compile_definitions`, and needed to be after the definition of the dosbox target.

johnnovak commented:
Nice improvement @shermp, I will test it later today. It's great you started using commit prefixes! I think `build: ` would be a bit better, though. I should add the list of preferred prefixes to our doco to ensure consistency.

shermp commented:
I changed the commit message prefix to `build:`, and moved the `add_compile_definitions` as requested.

shermp commented:
For those interested, the build time differences on my Ryzen 9700X when enabling `UseMultiToolTask` is as follows:
```
[driver] Build completed: 00:03:32.858
[driver] Build completed: 00:00:41.576
```
I cleaned, and deleted and reconfigured cmake cache between the two runs.

johnnovak reviewed (approved):
Nice improvement @shermp! 🎉

johnnovak commented:
Can we merge this @shermp ?

shermp commented:
Dunno if you tested, but feel free to merge.

johnnovak commented:
> Dunno if you tested, but feel free to merge.

Yeah I haven't A/B compared the speed improvements, just imported the project as a CMake project into MSVC and did a debug and release build. Everything seemed fine!
